src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main.go

```diff
@@ -63,33 +78,49 @@ func (cfg *Config) Setup() (cmd.Starter, error) {
 		flags.StringVar(&tlsConfig.KeyFile, "cert-key", "", "cert key")
 		flags.BoolVar(&tlsConfig.SkipVerify, "tls-skip-verify", false, "Skip TLS certificate verification - use it only for testing")
 		flags.BoolVar(&disableTLS, "disable-tls", false, "Disable TLS encryption - use it only for testing")
+		flags.BoolVar(&secureComms, "secure-comms", false, "Use SSH to secure communication between cluster and peer pods")
+		flags.StringVar(&secureCommsInbounds, "secure-comms-inbounds", "", "Inbound tags for secure communication tunnels")
+		flags.StringVar(&secureCommsOutbounds, "secure-comms-outbounds", "", "Outbound tags for secure communication tunnels")
 	})
 
-	if !disableTLS {
-		cfg.tlsConfig = &tlsConfig
-	}
-
 	cmd.ShowVersion(programName)
 
 	if showVersion {
 		cmd.Exit(0)
 	}
 
-	for path, obj := range map[string]interface{}{
-		cfg.configPath: &cfg.daemonConfig,
-	} {
-		if err := load(path, obj); err != nil {
-			return nil, err
+	if err := load(cfg.configPath, &cfg.daemonConfig); err != nil {
+		return nil, err
+	}
+
+	if secureComms {
+		ppssh.Singleton()
+		host, port, err := net.SplitHostPort(cfg.listenAddr)
+		if err != nil {
+			logger.Printf("Warning: illegal address %s is changed to 127.0.0.1:15150 since secure-comms is enabled.", cfg.listenAddr)
+		}
+		if host != "127.0.0.1" {
+			logger.Printf("Address %s is changed to 127.0.0.1:%s since secure-comms is enabled.", cfg.listenAddr, port)
+			cfg.listenAddr = "127.0.0.1:" + port
+		}
```

Comment:
Can these strings be declared as constant for easier readability and possible to verify the inbounds and outbounds to follow acceptable pattern?

Reply:
@bpradipt 
Do you mean making `B:KBS:8080` into a constant?
This would be:
const INBOUND_KBS_FOR_BOTH_ATTESTATION_AND_KUBERNETES_PHASES = "B:KBS:8080"

Today one can see the structure of the tag and it helps understand what it is, translating the entire tag to a single string name in more confusion imho.

Reply:
I meant only the individual strings K (KUBERNETES_PHASE)
A (ATTESTATION_PHASE), B - may be a good term to indicate both phases etc.  Converting entire string to constant is non intuitive.
I'm ok to leave it as is, but at least a comment in the code itself indicating what it means and what other options can be used will be useful from maintainability standpoint.

Reply:
ok, Now I get it, I will add constants for KUBERNETES_PHASE, ATTESTATION_PHASE, BOTH_PHASES

Reply:
I thin k using the constants is cumbersome in some cases  = e.g. doing `KUBERNETES_PHASE+"KATAAPI:0"` instead of `"K:KATAAPI:0"`

So how about doing: `"KUBERNETES_PHASE:KATAAPI:0"`
Ad adding the constants:
         KUBERNETES_PHASE  = "KUBERNETES_PHASE"
	ATTESTATION_PHASE = "ATTESTATION_PHASE"
	BOTH_PHASES       = "BOTH_PHASES"
so we can use them in comparisons etc.